web/src/hooks/usePluginAvailable.ts

```diff
@@ -0,0 +1,14 @@
+import * as React from 'react';
+import { useBoolean } from './useBoolean';
+
+export const usePluginAvailable = (pluginName: string): boolean => {
+  const [isPluginAvailable, togglePluginAvailable] = useBoolean(false);
+
+  React.useEffect(() => {
+    fetch(`/api/plugins/${pluginName}/plugin-manifest.json`).then(
+      (response) => response.status === 200 && togglePluginAvailable(),
```

Comment:
would this also work when we get a redirect `3XX` response?

Reply:
The response objects `status` parameter returns the status of the final hop, so a 301 or 302 redirect ending with a 200 will still work here. There is a separate `redirected` parameter to tell if the fetch has been redirected:

https://developer.mozilla.org/en-US/docs/Web/API/Response/redirected